app/lib/features/events/providers/event_providers.dart

```diff
@@ -186,28 +187,52 @@ final eventListSearchedProvider = FutureProvider.autoDispose
 final eventListQuickSearchedProvider =
     FutureProvider.autoDispose<List<ffi.CalendarEvent>>((ref) async {
   final searchTerm = ref.watch(quickSearchValueProvider);
+
+  final priotizeBookmarkedEvents = await priotizeBookmarked(
+    ref,
+    BookmarkType.events,
+    await ref.watch(allEventListProvider(null).future),
+    getId: (t) => t.eventId().toString(),
+  );
+
   return _filterEventBySearchTerm(
     searchTerm,
-    await ref.watch(allEventListProvider(null).future),
+    priotizeBookmarkedEvents,
   );
 });
 
 final eventListSearchedAndFilterProvider = FutureProvider.autoDispose
     .family<List<ffi.CalendarEvent>, String?>((ref, spaceId) async {
   //Declare filtered event list
-  final filteredEventList =
+
+  final bookmarkedEvents =
```

Comment:
why this? Why `await` on _all_ of these now if only some are needed in any case? I understand the other might be a bit harder to parse but this is executes unnecessary code (and `ref.watch`es for unnecessary changes).

I'd suggest reverting to the old swtich-statement.

Reply:
<img width="895" alt="Screenshot 2024-11-13 at 4 26 18 PM" src="https://github.com/user-attachments/assets/b4d72aab-472a-49e2-9d63-28f6b74abf0e">

Reply:
Also old switch statements doesn't sorting list by ongoing first then upcoming and at the last past event when switch case-all is selected.

Reply:
I can't follow. The error message you are showing me is for the new code ... what am I supposed to see there?

Reply:
As mentioned in above comment, your older code doesn't sorting list with ongoing follow by upcoming and past when ALL switch case selected.